content/en/blog/2021/stateful-set-made-easier/index.md

```diff
@@ -0,0 +1,226 @@
+tcp        0      0 0.0.0.0:2181            0.0.0.0:*               LISTEN
+tcp        0      0 10.96.7.7:3888          0.0.0.0:*               LISTEN
+tcp        0      0 127.0.0.1:2181          127.0.0.1:37412         TIME_WAIT
+tcp        0      0 127.0.0.1:2181          127.0.0.1:37486         TIME_WAIT
+tcp        0      0 127.0.0.1:2181          127.0.0.1:37456         TIME_WAIT
+tcp        0      0 127.0.0.1:2181          127.0.0.1:37498         TIME_WAIT
+tcp        0      0 127.0.0.1:2181          127.0.0.1:37384         TIME_WAIT
+tcp        0      0 127.0.0.1:2181          127.0.0.1:37514         TIME_WAIT
+tcp        0      0 127.0.0.1:2181          127.0.0.1:37402         TIME_WAIT
+tcp        0      0 127.0.0.1:2181          127.0.0.1:37434         TIME_WAIT
+tcp        0      0 127.0.0.1:2181          127.0.0.1:37526         TIME_WAIT
+tcp        0      0 127.0.0.1:2181          127.0.0.1:37374         TIME_WAIT
+tcp        0      0 127.0.0.1:2181          127.0.0.1:37442         TIME_WAIT
+tcp        0      0 127.0.0.1:2181          127.0.0.1:37464         TIME_WAIT
+{{< /text >}}
+
+There is nothing `ESTABLISHED` on port 2888 or 3888.  Next, let us get the ZooKeeper server status:
+
+{{< text bash yaml >}}
+$ kubectl exec my-release-zookeeper-1 -c zookeeper -- /opt/bitnami/zookeeper/bin/zkServer.sh status
+/opt/bitnami/java/bin/java
+ZooKeeper JMX enabled by default
+Using config: /opt/bitnami/zookeeper/bin/../conf/zoo.cfg
+Client port found: 2181. Client address: localhost. Client SSL: false.
+Error contacting service. It is probably not running.
+{{< /text >}}
+
+From the above output, you can see the ZooKeeper service is not functioning properly. Let us check the cluster configuration for one of the ZooKeeper pods:
+
+{{< text bash yaml >}}
+$ istioctl proxy-config cluster my-release-zookeeper-1 --port 3888 --direction inbound -o json
+[
+    {
+        "name": "inbound|3888||",
+        "type": "STATIC",
+        "connectTimeout": "10s",
+        "loadAssignment": {
+            "clusterName": "inbound|3888||",
+            "endpoints": [
+                {
+                    "lbEndpoints": [
+                        {
+                            "endpoint": {
+                                "address": {
+                                    "socketAddress": {
+                                        "address": "127.0.0.1",
+                                        "portValue": 3888
+                                    }
+                                }
+                            }
+                        }
+                    ]
+                }
+            ]
+        },
+...
+{{< /text >}}
+
+What is interesting here is that the inbound on port 3888 has `127.0.0.1` as its endpoint. This means that Envoy proxy before Istio 1.10 redirects the inbound traffic to the `loopback` interface, as described in the [upcoming networking change in 1.10 blog](/blog/2021/upcoming-networking-changes/).
```

Comment:
```suggestion
What is interesting here is that the inbound on port 3888 has `127.0.0.1` as its endpoint. This is because the Envoy proxy, in versions of Istio prior to 1.10, redirects the inbound traffic to the `loopback` interface, as described in the [our blog post about the change](/blog/2021/upcoming-networking-changes/).
```